data/src/main/java/com/example/data/db/dao/WeatherDataDao.kt

```diff
@@ -0,0 +1,18 @@
+package com.example.data.db.dao
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import com.example.data.db.entities.WeatherDataEntity
+
+
+@Dao
+interface WeatherDataDao {
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertWeatherData(weatherDataEntity: WeatherDataEntity)
+
+    @Query("SELECT * FROM weather_data WHERE city_name = :cityName AND date = :currentDate")
+    fun getWeatherDataFromDb(cityName: String, currentDate: String): WeatherDataEntity
```

Comment:
Fixed. I had to change RxJava3 on RxJava2 because room does not support RxJava3. From July 22, 2020 a new dependency `androidx.room:room-rxjava3:2.3.0-alpha02`  is available. I tried to use that but it was not working in my project